src/metabase/pulse/render/body.clj

```diff
@@ -512,14 +512,80 @@
       [:img {:style (style/style {:display :block :width :100%})
              :src   (:image-src image-bundle)}]]}))
 
+(defn- overlap
+  "calculate the overlap, a value between 0 and 1, of the ranges of 2 columns.
+  This overlap value can be checked against `axis-group-threshold` to determine when columns can reasonably share a y-axis.
+  Consider two ranges, with min and max values:
+
+   min-a = 0                                 max-a = 43
+     *-----------------------------------------*
+                                                      min-b = 52             max-b = 75
+                                                        *----------------------*
+  The overlap above is 0. The mirror case where col-b is entirely less than col-a, and also has 0 overlap.
+  Otherwise, overlap is calculated as follows:
+
+     min-a = 0                                 max-a = 43
+     *-----------------------------------------*
+     |     min-b = 8                           |             max-b = 59
+     |       *---------------------------------|---------------*
+     |       |                                 |               |
+     |       |- overlap-width = (- 43 8) = 35 -|               |
+     |                                                         |
+     |--------- max-width = (- 59 0) = 59 ---------------------|
+
+  overlap = (/ overlap-width max-width) = (/ 35 59) = 0.59
+
+  Another scenario, with a similar result may look as follows:
+
+     min-a = 0                                                 max-a = 59
+     *---------------------------------------------------------*
+     |     min-b = 8                         max-b = 43        |
+     |       *---------------------------------*               |
+     |       |                                 |               |
+     |       |- overlap-width = (- 43 8) = 35 -|               |
+     |                                                         |
+     |--------- max-width = (- 59 0) = 59 ---------------------|
+
+  overlap = (/ overlap-width max-width) = (/ 35 59) = 0.59
+"
+  [col-a col-b]
+  (let [[min-a min-b] (map #(get-in % [:fingerprint :type :type/Number :min]) [col-a col-b])
+        [max-a max-b] (map #(get-in % [:fingerprint :type :type/Number :max]) [col-a col-b])
+        [a b c d] (sort [min-a min-b max-a max-b])
+        max-width (- d a)
+        overlap-width (- c b)]
+    (if (or (and (< max-a min-b) (< max-a max-b))
+            (and (> min-a min-b) (> min-a max-b)))
+      ;; the smaller range is completely ahead of or behind the larger range
+       0
+      ;; there is some overlap, so what's the overlap of the whole width of both ranges?
+      (/ overlap-width max-width))))
```

Comment:
Sorry to be a diva about this, but I almost left the exact same comment on #556 that I left last week. I propose this, which I think makes the logic a little clearer (and prevents `overlap-width` from being a misnomer if the ranges don't overlap):

```clojure
(let [[min-a min-b]    (map #(get-in % [:fingerprint :type :type/Number :min]) [col-a col-b])
      [max-a max-b]    (map #(get-in % [:fingerprint :type :type/Number :max]) [col-a col-b])
      non-overlapping? (or (and (< max-a min-b) (< max-a max-b))
                           (and (> min-a min-b) (> min-a max-b)))]
   (if non-overlapping?
     0
     (let [[a b c d]     (sort [min-a min-b max-a max-b])
           max-width     (- d a)
           overlap-width (- c b)]
       (/ overlap-width max-width))))
```

Reply:
Not being a Diva at all, I'm really sorry I didn't address it prior.
I'll pop that change in, get tests green, and then merge. Thanks :)